### PR TITLE
KA-3: [node sample] Added error handling

### PR DESCRIPTION
**Pull request by Fleeck**
https://sandbox.tradeshift.com/#/apps/Kind.nodesample/
Error handling tab

### CODE REVIEW

**Comment by TimeBomb** [comment]
Had some issues with the new GitHub review system, sorry about that @Fleeck. You should be able to see all my comments now on your several PRs. Thanks again.

File: javascript/nodejs/browser/javascripts/home.controller.js [hunk 1]
```diff
@@ -195,16 +205,17 @@ app.controller('HomeCtrl', function($scope, $req, $window, $translate, $q) {
...
-    var array = [];
-    for (p in item){
-      array.push(item[p] + ''); // here we get array of strings
-    }
-    result.push(array);
-  });
-  return result; // array of arrays
-};
+      var result = [];
+      data.forEach(function(item){
```
Comment (Fleeck) [inline javascript/nodejs/browser/javascripts/home.controller.js]: updated here https://github.com/Tradeshift/tradeshift-app-samples/pull/49/commits/00235946d016e04a077138ecc394cc3657bf056e

File: javascript/nodejs/browser/javascripts/home.controller.js [hunk 2]
```diff
@@ -213,4 +224,13 @@ app.controller('HomeCtrl', function($scope, $req, $window, $translate, $q) {
         ts.ui.Notification.error(response.status + ' ' + response.statusText);
       })
     };
+
+    /* For error handling */
+    $scope.sendFailingRequest = function(status){
+      $req.sendFailingRequest(status).then(function(response){
+        // Nothing gonna be here, because it will fail
+      }, function(response){ // on error
```
Comment (Fleeck) [inline javascript/nodejs/browser/javascripts/home.controller.js]: @TimeBomb yeah, you're totally right here. Updated.

File: javascript/nodejs/browser/javascripts/app.js [hunk 3]
```diff
@@ -58,6 +58,13 @@ app.config(function ($locationProvider, $translateProvider) {
       },
       createTask: function () { // in case if you want to get your locales from server, currently not used
         return $http.post(url + 'tasks');
+      },
+      sendFailingRequest: function(status) {
+        if (status == 404){
```
Comment (Fleeck) [inline javascript/nodejs/browser/javascripts/app.js]: Changed

Comment (TimeBomb) [inline javascript/nodejs/browser/javascripts/app.js]: Looks like it was changed in the below `else if` but not in this `if` statement.

File: javascript/nodejs/browser/views/index.html [hunk 4]
```diff
@@ -310,6 +310,20 @@
       </div>
       <div data-ts="CompanyCard" id="home-card"></div>
     </div>
+    <div id="tab9" ng-show="showTab == 9">
```
Comment (TimeBomb) [inline javascript/nodejs/browser/views/index.html]: `===` over `==`

**Comment by TimeBomb** [comment]
Once PR feedback is addressed, 👍

**Comment by Fleeck** [comment]
Updated & squashed. Merging now.

**Pull request merged**